apps/site/assets/ts/stop/components/DeparturesAndMap.tsx

```diff
@@ -52,16 +53,25 @@ const DeparturesAndMap = ({
     });
   };
 
-  const viewAllRoutes: () => boolean = () => {
-    if (
-      !departureInfo.departureRoute &&
-      !departureInfo.departureDirectionId &&
-      !departureInfo.departureSchedules
-    ) {
-      return true;
+  const viewSelectedDeparture = !some(
+    Object.values(departureInfo),
+    isUndefined
+  );
+
+  const isSmallBreakpoint = useSMDown();
+  const refEl = useRef<HTMLDivElement>(null);
+
+  // prevent scrolling the page when in fullscreen "app" view
+  useLayoutEffect(() => {
+    if (isSmallBreakpoint && viewSelectedDeparture && refEl.current) {
+      disableBodyScroll(refEl.current);
+    } else {
+      clearAllBodyScrollLocks();
     }
-    return false;
-  };
+    return () => {
+      clearAllBodyScrollLocks();
+    };
+  }, [viewSelectedDeparture, isSmallBreakpoint]);
```

Comment:
[This blog post](https://epicreact.dev/why-you-shouldnt-put-refs-in-a-dependency-array/) explains this a bit. The eslint warning describes it decently well (example):

> React Hook React.useEffect has an unnecessary dependency: 'usernameInputRef.current'.
Either exclude it or remove the dependency array.
Mutable values like 'usernameInputRef.current' aren't valid dependencies because
mutating them doesn't re-render the component.